common/transport/DcgmIpc.cpp

```diff
@@ -579,8 +595,21 @@ void DcgmIpc::ConnectTcpAsyncImpl(DcgmIpcConnectTcp &tcpConnect)
     bufferevent_setcb(bev, DcgmIpc::StaticReadCB, NULL, DcgmIpc::StaticEventCB, this);
     bufferevent_enable(bev, EV_READ | EV_WRITE);
 
+    /* Allow IPv6 IPs to override family */
+    char buf[16];
+    sa_family_t family = AF_INET;
+    std::string str = tcpConnect.m_hostname;
+    if (str.size() > 1 && str[0] == '[' && str[str.size() - 1] == ']')
+    {
+        tcpConnect.m_hostname = str.substr(1, str.size() - 2);
+    }
+    if (inet_pton(AF_INET6, tcpConnect.m_hostname.c_str(), buf) > 0)
```

Comment:
Please clear buf before use:
```suggestion
    memset(buf, 0, sizeof(buf));
    if (inet_pton(AF_INET6, tcpConnect.m_hostname.c_str(), buf) > 0)
```